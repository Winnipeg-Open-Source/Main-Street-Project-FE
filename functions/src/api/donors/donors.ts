import { DONORS_COLLECTION } from '../../constants/collections';
import { getRouter } from '../../utils/api';

export default getRouter(DONORS_COLLECTION);
