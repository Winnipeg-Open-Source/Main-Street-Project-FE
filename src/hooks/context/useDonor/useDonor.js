import { useContext } from 'react';
import DonorsContext from 'context/Donors';

function useDonor (id) {
    const { donors, ...rest } = useContext(DonorsContext);
    const donor = donors && donors.find((donor) => donor.id == id);

    return {
        ...rest,
        donor,
    };
}

export default useDonor;
