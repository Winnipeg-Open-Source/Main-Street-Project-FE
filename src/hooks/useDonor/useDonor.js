import { useContext } from 'react';
import DonorsContext from 'context/Donors';

function useDonor (id) {
    const { donors, ...rest } = useContext(DonorsContext);
    const donor = donors && donors.find(donor => parseInt(donor.id) === parseInt(id));

    return {
        ...rest,
        donor,
    };
};

export default useDonor;
