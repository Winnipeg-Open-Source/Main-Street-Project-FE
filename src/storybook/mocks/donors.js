export const noDonors = {
    isLoading: false,
    donors: [],
};

export const singleDonor = {
    isLoading: false,
    donors: [
        {
            id: 1,
            name: 'Costco',
            address: '22 Road Lane',
            city: 'Winnipeg, MB',
            email: 'costco@costco.com',
            phoneNumber: '204-555-5555',
            lastDonationDate: '2019 December 20',
            donations: [
                {
                    donationDate: '2019 December 20',
                },
                {
                    donationDate: '2019 November 20',
                },
            ],
        },
    ],
};

export const multipleDonors = {
    isLoading: false,
    donors: [
        {
            id: 1,
            name: 'Costco',
            lastDonationDate: '2019 December 20',
        },
        {
            id: 2,
            name: 'Sobeys',
            lastDonationDate: '2019 December 20',
        },
        {
            id: 3,
            name: 'Jane Doe',
            lastDonationDate: '2019 December 20',
        },
    ],
};
