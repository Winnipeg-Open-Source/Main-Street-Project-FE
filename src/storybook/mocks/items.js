export const mockItemRequisition = [
    {
        id: 1,
        name: 'Paper Towel',
        quantity: 5,
        quantityName: 'Rolls',
    },
    {
        id: 2,
        name: 'Blueberries',
        quantity: 10,
        quantityName: 'Packages',
    },
    {
        id: 3,
        name: 'Steaks',
        quantity: 2,
    },
];