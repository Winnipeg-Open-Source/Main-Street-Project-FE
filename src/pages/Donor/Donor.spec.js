import React from 'react';
import renderWithThemeAndRouter from 'tests/helpers/renderWithThemeAndRouter';
import DonorPage from './Donor';
import useDonor from 'hooks/context/useDonor';
import { donor } from 'tests/mocks/donors';

jest.mock('hooks/context/useDonor');

describe ('DonorPage', () => {
    it ('renders without crashing', () => {
        useDonor.mockImplementation(() => ({
            isLoading: true,
            donor,
        }));

        const { asFragment } = renderWithThemeAndRouter(<DonorPage />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('renders loading state', () => {
        useDonor.mockImplementation(() => ({
            isLoading: true,
        }));
        
        const { asFragment } = renderWithThemeAndRouter(<DonorPage />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('has a donor', async () => {
        useDonor.mockImplementation(() => ({
            isLoading: false,
            donor,
        }));

        const { getByText } = renderWithThemeAndRouter(<DonorPage />);

        expect(getByText('Costco')).toBeInTheDocument();
        expect(getByText('22 Road Lane')).toBeInTheDocument();
        expect(getByText('Winnipeg, MB')).toBeInTheDocument();
        expect(getByText('email@email.com')).toBeInTheDocument();
        expect(getByText('email@email.com')).toBeInTheDocument();
        expect(getByText('204-555-5555')).toBeInTheDocument();
        expect(getByText('Donated: 2019 December 20')).toBeInTheDocument();
        expect(getByText('Donated: 2019 November 20')).toBeInTheDocument();
    });

    it ('does not find a donor', () => {
        useDonor.mockImplementation(() => ({
            isLoading: false,
            donor: null,
        }));

        const { getByText } = renderWithThemeAndRouter(<DonorPage />);

        expect(getByText('Donor Not Found')).toBeInTheDocument();
    });
});
