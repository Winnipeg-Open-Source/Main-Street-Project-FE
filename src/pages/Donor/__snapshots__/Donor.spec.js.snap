// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonorPage renders loading state 1`] = `
<DocumentFragment>
  <div
    class="sc-pkfPC haHNKR sc-psDhf iAENMo"
  >
    <div
      class="sc-AxirZ sc-AxhCb jTGKnC"
      width="1"
    >
      <div
        class="sc-AxirZ sc-AxhCb iEMUrT"
        width="1"
      >
        <div
          class="sc-AxhUy sc-qQxFs hINyDI"
          font-size="3"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 200px; height: 29px;"
            >
              ‌
            </span>
          </span>
        </div>
        <span>
          <span
            class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
            style="width: 100px; height: 29px;"
          >
            ‌
          </span>
        </span>
      </div>
      <div
        class="sc-AxirZ sc-pCPXO sc-qYUWV kPjGTf sc-oTzgz kLGYlD"
        color="background.lightest"
        width="1"
      >
        <div
          class="sc-AxhUy gpDHNu"
          color="text.darkest"
          font-size="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 80%;"
            >
              ‌
            </span>
          </span>
        </div>
        <div
          class="sc-AxhUy gpDHNu"
          color="text.darkest"
          font-size="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 80%;"
            >
              ‌
            </span>
          </span>
        </div>
        <div
          class="sc-AxhUy gpDHNu"
          color="text.darkest"
          font-size="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 80%;"
            >
              ‌
            </span>
          </span>
        </div>
        <div
          class="sc-AxhUy gpDHNu"
          color="text.darkest"
          font-size="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 80%;"
            >
              ‌
            </span>
          </span>
        </div>
      </div>
      <div
        class="sc-AxirZ sc-AxhCb sc-pBlxj cDPXBJ"
        width="1"
      >
        <div
          class="sc-AxirZ sc-pCPXO sc-qYUWV kPjGTf sc-pRgDJ eZkolW"
          color="background.lightest"
          width="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 25%;"
            >
              ‌
            </span>
          </span>
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 50%;"
            >
              ‌
            </span>
          </span>
        </div>
        <div
          class="sc-AxirZ sc-pCPXO sc-qYUWV kPjGTf sc-pRgDJ eZkolW"
          color="background.lightest"
          width="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 25%;"
            >
              ‌
            </span>
          </span>
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 50%;"
            >
              ‌
            </span>
          </span>
        </div>
        <div
          class="sc-AxirZ sc-pCPXO sc-qYUWV kPjGTf sc-pRgDJ eZkolW"
          color="background.lightest"
          width="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 25%;"
            >
              ‌
            </span>
          </span>
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 50%;"
            >
              ‌
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DonorPage renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-pkfPC haHNKR sc-psDhf iAENMo"
  >
    <div
      class="sc-AxirZ sc-AxhCb jTGKnC"
      width="1"
    >
      <div
        class="sc-AxirZ sc-AxhCb iEMUrT"
        width="1"
      >
        <div
          class="sc-AxhUy sc-qQxFs hINyDI"
          font-size="3"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 200px; height: 29px;"
            >
              ‌
            </span>
          </span>
        </div>
        <span>
          <span
            class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
            style="width: 100px; height: 29px;"
          >
            ‌
          </span>
        </span>
      </div>
      <div
        class="sc-AxirZ sc-pCPXO sc-qYUWV kPjGTf sc-oTzgz kLGYlD"
        color="background.lightest"
        width="1"
      >
        <div
          class="sc-AxhUy gpDHNu"
          color="text.darkest"
          font-size="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 80%;"
            >
              ‌
            </span>
          </span>
        </div>
        <div
          class="sc-AxhUy gpDHNu"
          color="text.darkest"
          font-size="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 80%;"
            >
              ‌
            </span>
          </span>
        </div>
        <div
          class="sc-AxhUy gpDHNu"
          color="text.darkest"
          font-size="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 80%;"
            >
              ‌
            </span>
          </span>
        </div>
        <div
          class="sc-AxhUy gpDHNu"
          color="text.darkest"
          font-size="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 80%;"
            >
              ‌
            </span>
          </span>
        </div>
      </div>
      <div
        class="sc-AxirZ sc-AxhCb sc-pBlxj cDPXBJ"
        width="1"
      >
        <div
          class="sc-AxirZ sc-pCPXO sc-qYUWV kPjGTf sc-pRgDJ eZkolW"
          color="background.lightest"
          width="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 25%;"
            >
              ‌
            </span>
          </span>
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 50%;"
            >
              ‌
            </span>
          </span>
        </div>
        <div
          class="sc-AxirZ sc-pCPXO sc-qYUWV kPjGTf sc-pRgDJ eZkolW"
          color="background.lightest"
          width="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 25%;"
            >
              ‌
            </span>
          </span>
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 50%;"
            >
              ‌
            </span>
          </span>
        </div>
        <div
          class="sc-AxirZ sc-pCPXO sc-qYUWV kPjGTf sc-pRgDJ eZkolW"
          color="background.lightest"
          width="1"
        >
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 25%;"
            >
              ‌
            </span>
          </span>
          <span>
            <span
              class="react-loading-skeleton css-1x45xix-skeletonStyles-Skeleton"
              style="width: 50%;"
            >
              ‌
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
